tests/WebServer.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace DDTrace\Tests;
+
+use Symfony\Component\Process\Process;
+
+/**
+ * A controllable php server running in a separate process.
+ */
+class WebServer
+{
+    /**
+     * Symfony Process object managing the server
+     *
+     * @var Process
+     */
+    private $process;
+
+    /**
+     * @var string
+     */
+    private $host;
+
+    /**
+     * @var int
+     */
+    private $port;
+
+    /**
+     * @var string
+     */
+    private $indexFile;
+
+    /**
+     * @var array
+     */
+    private $envs = [];
+
+    private $defaultEnvs = [
+        'DD_TRACE_AGENT_PORT' => 80,
+        'DD_AGENT_HOST' => 'request_replayer',
+    ];
+
+    /**
+     * @param string $indexFile
+     * @param string $host
+     * @param int $port
+     */
+    public function __construct($indexFile, $host = '0.0.0.0', $port = 80)
+    {
+        $this->indexFile = $indexFile;
+        $this->host = $host;
+        $this->port = $port;
+    }
+
+    /**
+     *
+     */
+    public function start()
+    {
+        $host = $this->host;
+        $port = $this->port;
+        $indexFile = $this->indexFile;
+        $indexFileDir = dirname($indexFile);
+        $cmd = "php -dlog_errors=on -derror_log='$indexFileDir/error.log' -S $host:$port $indexFile";
+        $envs = $this->getSerializedEnvsForCli();
+        $processCmd = "$envs exec $cmd";
+        $this->process = new Process($processCmd);
+        $this->process->start();
+        usleep(100000);
```

Comment:
Does `start()` not block until it is ready? I'm just not understanding why we need the `usleep()` here. :)

Reply:
Good point. Actually not since the server is run through a Symfony `Process` object  which executed the  command  in a sub process and is not blocking: https://symfony.com/doc/current/components/process.html#usage